### PR TITLE
Make certain utils generic for future projects

### PR DESCRIPTION
**Pull request by NickGoog**
Make temporary file naming utils generic.
Make binary running logic generic.
Fix an issue where unit tests fail if `cert_provider_command` is present in `.boto`.

### CODE REVIEW

File: gslib/tests/test_temporary_file_util.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+from gslib import storage_url
+from gslib.tests import testcase
+from gslib.utils import temporary_file_util
+
+
+class TestTemporaryUtil(testcase.GsUtilUnitTestCase):
+  """Test temporary file utils."""
+
+  def test_gets_temporary_file_name(self):
```
Comment (rrauber) [inline gslib/tests/test_temporary_file_util.py:30]: I think gsutil tends to use lower camel case for test names. I don't think it's 100% consistent but may be good to change this to match the other test file.

File: gslib/utils/execution_util.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import division
+from __future__ import unicode_literals
+
+import subprocess
+
+
+def execute_external_command(command_and_flags):
```
Comment (rrauber) [inline gslib/utils/execution_util.py:25]: I think this should be upper camel case

**Review by rrauber — CHANGES_REQUESTED** [review]

File: gslib/tests/test_temporary_file_util.py (hunk 1 above)
Comment (NickGoog) [reply gslib/tests/test_temporary_file_util.py]: Yes, great catch.

File: gslib/utils/execution_util.py (hunk 2 above)
Comment (NickGoog) [reply gslib/utils/execution_util.py]: Good catch

**Review by rrauber — APPROVED** [review]

**Pull request merged**